src/modeler/AutoRest.Swagger/Validation/CollectionObjectPropertiesNamingValidation.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+using AutoRest.Swagger.Model.Utilities;
+using System.Linq;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class CollectionObjectPropertiesNamingValidation : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
+    {
+        private readonly Regex ListRegex = new Regex(@".+_List([^_]*)$", RegexOptions.IgnoreCase);
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Dictionary<string, Operation>> entity, RuleContext context)
+        {
+            // get all operation objects that are either of get or post type
+            var serviceDefinition = (ServiceDefinition)context.Root;
+            var listOperations = entity.Values.SelectMany(opDict => opDict.Where(pair => pair.Key.ToLower().Equals("get") || pair.Key.ToLower().Equals("post")));
+            foreach (var opPair in listOperations)
+            {
+                // if the operation id is not of type _list* or does not return an array type, skip
+                if (!ListRegex.IsMatch(opPair.Value.OperationId) || !ValidationUtilities.IsXmsPageableOrArrayResponseOperation(opPair.Value, serviceDefinition))
+                {
+                    continue;
+                }
+                
+                var collType = opPair.Value.Responses["200"]?.Schema?.Reference?.StripDefinitionPath();
+                // if no response type defined skip
+                if (collType == null)
+                {
+                    continue;
+                }
+
+                var collTypeDef = serviceDefinition.Definitions[collType];
+                // if collection object has 2 properties or less (xmspageable objects can have the nextlink prop)
```

Comment:
minor: I'd follow `x-ms-pageable` exact name of extension so that it becomes easier to search the usage even in comments :)